rclcpp/test/rclcpp/test_subscription.cpp

```diff
@@ -461,6 +461,33 @@ TEST_P(TestSubscriptionInvalidIntraprocessQos, test_subscription_throws) {
   }
 }
 
+/*
+   Testing subscription with invalid use_intra_process_comm
+ */
+TEST_P(TestSubscriptionInvalidIntraprocessQos, test_subscription_throws_intraprocess) {
+  // Adds more coverage
+  rclcpp::SubscriptionOptionsWithAllocator<std::allocator<void>> options;
+  options.use_intra_process_comm = static_cast<rclcpp::IntraProcessSetting>(5);
+
+  initialize();
+  rclcpp::QoS qos = GetParam().qos;
+  using test_msgs::msg::Empty;
```

Comment:
Since we only use this one time below, I'd rather just spell it out down where it is used.  But that's just a suggestion, not a requirement.